azurelinuxagent/common/protocol/wire.py

```diff
@@ -858,47 +859,56 @@ def check_wire_protocol_version(self):
-        ret = None
+        """
+        Calls host_func on host channel and accounts for stale resource (ResourceGoneError or InvalidContainerError).
+        If stale, it refreshes the goal state and retries host_func.
+        This method can throw, so the callers need to handle that.
+        """
         try:
             ret = host_func()
-        except (ResourceGoneError, InvalidContainerError) as e:  # pylint: disable=C0103
+            if ret is None or not ret:
+                raise Exception("Request failed using the host channel.")
+
+            return ret
+        except (ResourceGoneError, InvalidContainerError) as error:
             host_plugin = self.get_host_plugin()
-            old_container_id = host_plugin.container_id
-            old_role_config_name = host_plugin.role_config_name
 
+            old_container_id, old_role_config_name = host_plugin.container_id, host_plugin.role_config_name
             msg = "[PERIODIC] Request failed with the current host plugin configuration. " \
                   "ContainerId: {0}, role config file: {1}. Fetching new goal state and retrying the call." \
-                  "Error: {2}".format(old_container_id, old_role_config_name, ustr(e))
+                  "Error: {2}".format(old_container_id, old_role_config_name, ustr(error))
             logger.periodic_info(logger.EVERY_SIX_HOURS, msg)
 
             self.update_host_plugin_from_goal_state()
 
-            new_container_id = host_plugin.container_id
-            new_role_config_name = host_plugin.role_config_name
+            new_container_id, new_role_config_name = host_plugin.container_id, host_plugin.role_config_name
             msg = "[PERIODIC] Host plugin reconfigured with new parameters. " \
                   "ContainerId: {0}, role config file: {1}.".format(new_container_id, new_role_config_name)
             logger.periodic_info(logger.EVERY_SIX_HOURS, msg)
 
             try:
                 ret = host_func()
-                if ret:
-                    msg = "[PERIODIC] Request succeeded using the host plugin channel after goal state refresh. " \
-                          "ContainerId changed from {0} to {1}, " \
-                          "role config file changed from {2} to {3}.".format(old_container_id, new_container_id,
-                                                                             old_role_config_name, new_role_config_name)
-                    add_periodic(delta=logger.EVERY_SIX_HOURS,
-                                 name=AGENT_NAME,
-                                 version=CURRENT_VERSION,
-                                 op=WALAEventOperation.HostPlugin,
-                                 is_success=True,
-                                 message=msg,
-                                 log_event=True)
 
-            except (ResourceGoneError, InvalidContainerError) as e:  # pylint: disable=C0103
+                if ret is None or not ret:
```

Comment:
The `ret is None` clause is redundant, no? `not ret` already covers the `ret == None` case.

Reply:
This is intentionally explicit to make the code more readable and signal clearly that `ret` can be both `None` or False.

Reply:
I don't think this is a blocker for the PR, but it might be better to have, e.g.,
```python
if ret in (None, False):
    ...
```

Not necessary, but a bit nicer way to make the code explicit.

This also has the benefit (however small) of ignoring `ret == [], {}` for example.

Reply:
This does look cleaner, let me make that change. Thanks!